common/model/transfer.c

```diff
@@ -21,6 +21,10 @@ retcode_t transfer_message_set_trytes(transfer_t* tf, tryte_t const* const tryte
 }
 
 retcode_t transfer_message_set_string(transfer_t* tf, char const* const str) {
+  if (strlen(str) == 0) {
```

Comment:
You can simply rewrite as `if (!str) {`, which is shorter and faster.

Reply:
this check is both for NULL pointer and string length. cannot use `if (!str) {` in this case.

Reply:
Use `if (!str || !*str)` instead.

Reply:
don't see a reason for it, but here might misleading and don't need to call `strlen` twice.